src/transformStateWithClones.js

```diff
@@ -8,6 +8,34 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateClone = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action['type']) {
+      case 'addProperties':
+        Object.assign(stateClone, action['extraData']);
+        break;
+      case 'removeProperties':
+        for (const key of action['keysToRemove']) {
```

Comment:
[CODE KNOWLEDGE]: Remember, if property key is a variable - use brackets object[key]. If it's called key - use dot access object.key.